lib/jsdom/browser/Window.js

```diff
@@ -182,24 +188,58 @@ function Window(options) {
   ///// METHODS
 
   let latestTimerId = 0;
+  let latestAnimationFrameCallbackId = 0;
 
   this.setTimeout = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setTimeout, clearTimeout, ++latestTimerId, fn, ms, timers, args);
   };
   this.setInterval = function (fn, ms) {
     const args = [];
     for (let i = 2; i < arguments.length; ++i) {
       args[i - 2] = arguments[i];
     }
-    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, args);
+    return startTimer(window, setInterval, clearInterval, ++latestTimerId, fn, ms, timers, args);
+  };
+  this.clearInterval = stopTimer.bind(this, timers);
+  this.clearTimeout = stopTimer.bind(this, timers);
+
+  if (this._pretendToBeVisual) {
+    this.requestAnimationFrame = fn => {
+      const hr = hrtime(windowInitialized);
+      const hrInNano = hr[0] * 1e9 + hr[1];
+      const fps = 1000 / 60;
+      const hrInMicro = hrInNano / 1e6;
+
+      return startTimer(
+        window,
+        setTimeout,
+        clearTimeout,
+        ++latestAnimationFrameCallbackId,
+        fn,
+        fps,
+        animationFrameCallbacks,
+        [hrInMicro]
+      );
+    };
+    this.cancelAnimationFrame = stopTimer.bind(this, animationFrameCallbacks);
+  } else {
+    // TODO: Should we throw to tell users about `pretendToBeVisual`?
+    // Also, should we somehow allow this to be polyfilled byt code using feature detection?
```

Comment:
I'd prefer if we just don't define them at all rather than no-op. I'm guessing this will lead to really hard-to-debug behavior for users, where the callback silently never gets called, and it's probably non-obvious why.

Maybe we can add a getter which prints a warning or something?

Reply:
I think not having the functions at all seems better.

Reply:
Cool, I removed them